### ISSUE TITLE
Layer kwarg ignored for fiona.open with Bytes files

### ISSUE THREAD

**Issue by owlas** [Toblerity/Fiona#781]
## Expected behavior and actual behavior.

*Expected*: Opening a Bytes stream with `fiona.open` and a specific named layer returns a collection for the named layer.

*Actual*: Returns the first (default layer) as specified by `fiona.listlayers`
## Steps to reproduce the problem.

```python
# Where afile is any file and mylayer is NOT the default layer
with fiona.open('afile', layer='mylayer') as f:
    print(f.name)

with open('afile', 'rb') as f:
    with fiona.open(f, layer='mylayer') as ff:
        print(ff.name)
```

## Operating system

Linux: 5.2.1-arch1-1-ARCH

## Fiona and GDAL version and provenance

Fiona==1.8.6 via pip 19.0.3

**Comment by snorfalorpagus** [comment]
I can reproduce this and have found a solution. Putting together a PR.

**Comment by owlas** [comment]
So fast! Looks like a quick fix. I'll test locally tomorrow

**Comment by chriselion** [comment]
Sorry for replying on a long-closed issue, but is this ever going to make it into a released version? The latest release (1.8.20) still doesn't pass the args to `memfile.open`.
